source/common/formatter/substitution_formatter.cc

```diff
@@ -1847,9 +1847,12 @@ GrpcStatusFormatter::GrpcStatusFormatter(const std::string& main_header,
     : HeaderFormatter(main_header, alternative_header, max_length), format_(format) {}
 
 absl::optional<std::string> GrpcStatusFormatter::format(
-    const Http::RequestHeaderMap&, const Http::ResponseHeaderMap& response_headers,
+    const Http::RequestHeaderMap& request_headers, const Http::ResponseHeaderMap& response_headers,
     const Http::ResponseTrailerMap& response_trailers, const StreamInfo::StreamInfo& info,
     absl::string_view, AccessLog::AccessLogType) const {
+  if (!Grpc::Common::isGrpcRequestHeaders(request_headers)) {
```

Comment:
Thanks you for your change. It looks good. However given that it is a user visible change it requires a runtime feature protection. If somebody has production monitoring that was affected by this change, they could temporarily disable the new behavior until they adjust. Here is a good example of how to add this: https://github.com/envoyproxy/envoy/pull/26764

Reply:
@yanavlasov , Thanks for your suggestion, it make sense. 👍 
I like this protection, will add this change.